### PR TITLE
Setup Dependabot and update actions

### PR DESCRIPTION
**Pull request by mmorel-35**
* Bump actions/download-artifact from 3 to 4.1.8 in /actions/restore_artifacts
* Bump actions/upload-artifact from 3 to 4.3.4 in /actions/save_artifacts
* Bump actions/cache from 3 to 4.0.2 in /actions/setup_environment
* Bump docker/setup-buildx-action in /actions/setup_environment
* Bump docker/setup-qemu-action from 1 to 3.2.0 in /actions/setup_environment

### CODE REVIEW

File: actions/restore_artifacts/action.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ runs:
   using: composite
   steps:
     - name: Download all workflow run artifacts
-      uses: actions/download-artifact@v3
+      uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
```
Comment (Nexucis) [inline actions/restore_artifacts/action.yml:9]: why are you using the commit ID and not the tag ?

Comment (mmorel-35) [reply actions/restore_artifacts/action.yml:9]: That's OpenSSF best practices recommendation. 
See https://github.com/ossf/scorecard/blob/68b5799b1abe0370790c56412cffff795a2faae0/docs/checks.md#pinned-dependencies

**Review by SuperQ — APPROVED** [review]

File: actions/restore_artifacts/action.yml (hunk 1 above)
Comment (Nexucis) [reply actions/restore_artifacts/action.yml:9]: nice, thanks for the explanations. LGTM

**Review by Nexucis — APPROVED** [review]

**Comment by Nexucis** [comment]
Thank you for the setup @mmorel-35 !

**Pull request merged**